File: fem/C++/openmp-target/auto.def/src/solve11.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include "pfem_util.h"

extern FILE *fp_log;
//extern void CG(); 
extern void  CG  (
	   KINT NP,KINT NPLU,KREAL D[],
	   KREAL AMAT[],KINT indexLU[], KINT itemLU[],
	   KREAL B[],KREAL X[],KREAL RESID,KINT ITER, KINT *ERROR); 
void SOLVE11()
{
  /*
  int i,j,k,ii,L;
  int  ICFLAG=0;
  CHAR_LENGTH BUF;
  */
  int  ERROR;
  
/**
   +------------+
   | PARAMETERs |
   +------------+
**/
  ITER      = pfemIarray[0];
  RESID     = pfemRarray[0];

/**
   +------------------+
   | ITERATIVE solver |
   +------------------+
**/
  CG (NP,NPLU, D, AMAT, indexLU, itemLU, B, X, RESID, ITER, &ERROR);
  ITERactual= ITER;
}

File: fem/F/do-concurrent/auto.def/src/solve11.f
      module SOLVER11

      contains
        subroutine SOLVE11

        use pfem_util
        use solver_CG
        implicit none

C        integer :: ERROR, ICFLAG
C        character(len=char_length) :: BUF
C        data ICFLAG/0/
        integer :: ERROR

!C
!C +------------+
!C | PARAMETERs |
!C +------------+
!C===
        ITER      = pfemIarray(1)
        RESID     = pfemRarray(1)
!C===

!C
!C +------------------+
!C | ITERATIVE solver |
!C +------------------+
!C===
        call CG                                                         &
     &  ( NP, NPLU, D, AMAT, indexLU, itemLU, B, X, RESID, ITER, ERROR )

      ITERactual= ITER
!C===

      end subroutine SOLVE11
      end module SOLVER11
